api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java

```diff
@@ -100,13 +101,13 @@ public HighFrequencyDetector() {
     List<Metric> newList = new ArrayList<>();
 
     // Groups metrics by metric hash code and finds times deltas of ordered data points.
-    // {hashcode -> -1, hashcode1 -> 10}
-    Map<Integer, Integer> groupedMetrics =
+    // {metric1 -> -1, metric2 -> 10}
+    Map<Metric, Integer> groupedMetrics =
         metrics.stream()
             .sorted(Comparator.comparing(Metric::getTime))
             .collect(
                 Collectors.groupingBy(
-                    Metric::hashCode,
+                    Function.identity(),
```

Comment:
What is the reasoning for switching to Function.identity() over metric hash code?

Reply:
My guess is, because we are now using metric.

Reply:
correct